### DISCUSSION TITLE
How to decode JWT token from an Express.js server?

### DISCUSSION THREAD

**Discussion by musjj** [nextauthjs/next-auth#10865] [Help]
I have a Next.js app that rewrites some API requests back to an Express.js backend. I want to be able to verify & decode the JWT token in the Express.js backend, how do I do this?

There's the `getToken` function which seems to be exactly what I'm looking for:

https://authjs.dev/reference/core/jwt#gettoken
https://next-auth.js.org/tutorials/securing-pages-and-api-routes#using-gettoken

But I can't figure out how to use this in an Express.js route. Does anyone have any working examples/snippets?

**Comment by kornilowds** [comment]
There is a `decode` function from `@auth/core/jwt`.

And we can use like:
```
await jwt.decode({
  token: <your-auth-js-jwt-token>
  secret: <AUTH_SECRET>
  salt: 'authjs.session-token',
}),
```
It works in Nextjs side, but in my Nest (express) server I've got `Package subpath './jwt' is not defined by "exports"` error

I also tried use import from `next-auth/jwt`

**Reply by musjj** [reply]
I've managed to solve it like this:

```typescript
await getToken({
  req: { headers: { cookie: /* your cookie */ } },
  secret: process.env.AUTH_SECRET,
  secureCookie: process.env.NODE_ENV === "production",
  salt:
    process.env.NODE_ENV === "production"
      ? "__Secure-authjs.session-token"
      : "authjs.session-token",
});
```

I also had your problem and I think it's related to ESM vs CJS (Auth.js is ESM-only). Try using `"type": "module"` in your `package.json`. I personally use `tsup` with `format: ["cjs"]` and `noExternal: [/.*/]` which transpiles all your dependencies into a single CJS file. It's kind of hacky but it helps solves problems like this (I think).
